core/helper_api.php

```diff
@@ -725,3 +725,36 @@ function helper_url_combine( $p_page, $p_query_string ) {
 
 	return $t_url;
 }
+
+/**
+ * Generate a hash to be used with dynamically generated content that is expected
+ * to be cached by the browser. This hash can be used to differentiate the generated
+ * content when it may be different based on some runtime attributes like: current user,
+ * project or language.
+ * An optional custom string can be provided to be added to the hash, for additional
+ * differentiating criteria, but this string must be already prepared by the caller.
+ *
+ * @param array $p_runtime_attrs    Array of attributes to be calculated from current session.
+ *                                  possible values: 'user', 'project', 'lang'
+ * @param type $p_custom_string     Additional string provided by the caller
+ * @return string                   A hashed md5 string
+ */
+function helper_generate_cache_key( array $p_runtime_attrs = [], $p_custom_string = '' ) {
+	$t_key = $p_custom_string;
```

Comment:
Not needed if you change the following line to
`$t_key = $p_custom_string . '+V' . MANTIS_VERSION;`